lib/tools/system-commands.js

```diff
@@ -80,71 +60,55 @@ systemCallMethods.connect = async function connect () {
-      try {
-        await comp.start(null, PROCESS_INIT_TIMEOUT);
-      } catch (e) {
-        throw new Error(`Cannot start ${IDB_COMPANION_EXECUTABLE}. Original error: ${e.message}`);
-      }
-      await B.delay(300);
-      if (!comp.isRunning) {
-        throw new Error(`${IDB_COMPANION_EXECUTABLE} has failed to start`);
-      }
-      log.info(`${IDB_COMPANION_EXECUTABLE} has been started with args ${JSON.stringify(args)}`);
-    } else {
-      log.info(`${IDB_COMPANION_EXECUTABLE} is already running for the device '${this.udid}'`);
-    }
-
-    const daemon = new SubProcess(IDB_EXECUTABLE, buildDaemonArgs(this.executable));
-    let daemonStderr = '';
-    daemon.on('output', (stdout, stderr) => {
-      if (isStartupMonitorEnabled && stderr) {
-        daemonStderr += `${stderr}\n`;
-      }
-    });
-    try {
-      await daemon.start(null, PROCESS_INIT_TIMEOUT);
-      await B.delay(300);
-    } catch (ign) {}
-    if (daemon.isRunning) {
-      log.info(`${IDB_EXECUTABLE} daemon is running on port ${this.port}`);
-    } else {
-      if (!daemonStderr.includes('address already in use')) {
-        throw new Error(`${IDB_EXECUTABLE} daemon has failed to start: ${daemonStderr}`);
-      }
-      log.info(`${this.port} is already in use. Assuming it is used by ${IDB_EXECUTABLE} daemon`);
-    }
-
-    const {stdout} = await tpExec(IDB_EXECUTABLE, ['list-targets', '--json']);
-    if (!fixOutputToArray(stdout).some((x) => _.toUpper(x.udid) === _.toUpper(this.udid))) {
-      throw new Error(`The device '${this.udid}' is unknown. Only these devices are known ` +
-        `to ${IDB_EXECUTABLE}: ${stdout}`);
-    }
-
     try {
       await waitForCondition(async () => {
         try {
+          const daemon = new SubProcess(IDB_EXECUTABLE, buildDaemonArgs(this.executable));
+          let daemonOutput = '';
+          daemon.on('output', (stdout, stderr) => {
+            if (isStartupMonitorEnabled && _.trim(stdout || stderr)) {
+              daemonOutput += `[daemon] ${stdout || stderr}\n`;
+            }
+          });
+          try {
+            await daemon.start(null, PROCESS_INIT_TIMEOUT);
+            await B.delay(300);
+          } catch (ign) {}
+          if (daemon.isRunning) {
+            log.debug(`${IDB_EXECUTABLE} daemon started on port ${this.executable.port || DEFAULT_IDB_PORT}`);
+          } else {
+            if (!daemonOutput.includes('address already in use')) {
+              const message = `${IDB_EXECUTABLE} daemon has failed to start: ${daemonOutput}`;
+              log.warn(message);
+              throw new Error(message);
+            }
+            log.debug(`The port ${this.executable.port || DEFAULT_IDB_PORT} is already in use. ` +
+              `Assuming it is used by ${IDB_EXECUTABLE} daemon`);
+          }
           await tpExec(IDB_EXECUTABLE, ['describe', '--udid', this.udid]);
           return true;
         } catch (e) {
+          if (e.stderr) {
+            log.debug(e.stderr);
```

Comment:
`log.error`?

Reply:
I don't want to make it confusing for the client, since it is not guaranteed that stderr contains the failure reason. Also, there might be too many irrelevant errors in the log after several retries